source/extensions/filters/http/rbac/rbac_filter.cc

```diff
@@ -56,6 +56,13 @@ absl::Status ActionValidationVisitor::performDataInputValidation(
   return absl::InvalidArgumentError(fmt::format("RBAC HTTP filter cannot match on '{}'", type_url));
 }
 
+Http::Code getDenyStatus(uint32_t deny_status) {
+  if (deny_status == 0) {
+    return Http::Code::Forbidden;
+  }
+  return static_cast<Http::Code>(deny_status);
```

Comment:
This is undefined behavior since you're casting from `uint32_t` to enum class over `uint16_t`. Maybe use https://github.com/envoyproxy/envoy/blob/e543e1c5dae6de3182d19c1b208ee4d1ee1b2fe7/api/envoy/type/v3/http_status.proto#L138?